torch/optim/optimizer.py

```diff
@@ -288,6 +289,16 @@ def wrapper(*args, **kwargs):
 
         return wrapper
 
+    @staticmethod
+    def _group_tensors_by_device_and_dtype(tensorlistlist, with_indices=False):
```

Comment:
could you add a comment for why this is necessary/explain that when compiling, the grouping will automatically happen at the…inductor? triton? level